GitHub.Common/Web/GitHubHttpClientBase.cs

```diff
@@ -0,0 +1,302 @@
+﻿// Copyright (c) Microsoft Corporation.
```

Comment:
Extracted this GitHubHttpClientBase class host the functions used by the authentication module and other modules. Leaving a function GetConditionalViaETagAsync() in the GitHubHttpClient.cs which is not used by authentication to reduce dragging more files into the common project.